### PR TITLE
Specialize templates to get fully qualified name

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #27 

`get_fully_qualified_name()` is not correct for the lifecycle component, so this needs specialization until we refactor the component interface.

## Review guidelines

Estimated Time of Review: 2 minutes

### CODE REVIEW

**Review by eeberhard — CHANGES_REQUESTED** [review]
NOT YET

File: source/modulo_components/include/modulo_components/ComponentInterface.hpp [hunk 1]
```diff
@@ -1256,15 +1256,24 @@ inline state_representation::CartesianPose ComponentInterface<NodeT>::lookup_tra
...
+template<>
+inline void ComponentInterface<rclcpp_lifecycle::LifecycleNode>::publish_predicate(const std::string& name) {
+  modulo_component_interfaces::msg::Predicate message;
+  message.component = this->get_node_base_interface()->get_fully_qualified_name();
+  message.predicate = name;
+  message.value = this->get_predicate(name);
+  this->predicate_publisher_->publish(message);
+}
+
 template<class NodeT>
```
Comment (domire8) [inline source/modulo_components/include/modulo_components/ComponentInterface.hpp:1277]:
```suggestion
template<class NodeT>
```

File: source/modulo_components/include/modulo_components/ComponentInterface.hpp [hunk 2]
```diff
@@ -1256,15 +1256,24 @@ inline state_representation::CartesianPose ComponentInterface<NodeT>::lookup_tra
   return result;
 }
 
-template<class NodeT>
-inline void ComponentInterface<NodeT>::publish_predicate(const std::string& name) {
+template<>
+inline void ComponentInterface<rclcpp::Node>::publish_predicate(const std::string& name) {
   modulo_component_interfaces::msg::Predicate message;
   message.component = this->get_fully_qualified_name();
```
Comment (domire8) [inline source/modulo_components/include/modulo_components/ComponentInterface.hpp:1262]:
```suggestion
template<typename NodeT>
inline void ComponentInterface<NodeT>::publish_predicate(const std::string& name) {
  modulo_component_interfaces::msg::Predicate message;
  message.component = this->get_node_base_interface()->get_fully_qualified_name();
```

**Review by eeberhard — APPROVED** [review]

**Comment by domire8** [comment]
I think this is now what you were looking for. The name of the predicate is `/<namespace>/<node_name>` or just `/<node_name>` like that (tested)

**Review by eeberhard — APPROVED** [review]

**Pull request merged**